### ISSUE TITLE
prettyPrint terminates with newline / no toString option

### ISSUE THREAD

**Issue by mn200** [polyml/polyml#99]
It seems that if I want to print to a string, I have to either reimplement Oppen, or write:
````
fun pp_to_string w f x =
  let
    val sbuf = ref [] : string list ref
    fun app s = (sbuf := s :: !sbuf)
  in
    prettyPrint (app,w) (f x);
    String.concat (List.rev (!sbuf))
  end
````
but then the current implementation of `prettyPrint` is guaranteed to put an extra newline as its last write to the output. 

It seems ugly to have to catch that final newline and then drop it.

**Comment by dcjm** [comment]
There is of course `PolyML.makestring` that will turn an arbitrary data structure into a string without any final newline.  It's a bit difficult to be sure whether this will do what you want.

**Comment by mn200** [comment]
It’s not great - the `pretty` value hasn’t necessarily come from a built-in/installed pretty-printer

**Comment by dcjm** [comment]
Pretty printing takes the `pretty` value and turns it into formatted text.  If it has to break a line it inserts a newline.  The idea is that the output is a series of lines each of which is terminated with a newline.  It seems to me to be inconsistent not to put a newline at the end of the last line.

**Comment by mn200** [comment]
I agree that this is a reasonable spec for `prettyPrint`.  However, if I want to turn the AST of something like `x + y` into a string so that I can embed it into something else (an error message, say), the final newline is not helpful.  I guess I'm asking for a version of `makeString` that takes a `pretty` rather than an `'a`.
